driver-core/src/main/com/mongodb/MongoClientSettings.java

```diff
@@ -477,12 +477,23 @@ public Builder uuidRepresentation(final UuidRepresentation uuidRepresentation) {
         /**
          * Sets the auto-encryption settings
          *
+         * A separate, internal {@code MongoClient} is created if any of the following are true:
+         *
+         * <ul>
+         *    <li>{@code AutoEncryptionSettings.keyVaultClient} is not passed</li>
+         *    <li>{@code AutoEncryptionSettings.bypassAutomaticEncryption} is {@code false}</li>
+         * </ul>
+         *
+         * If an internal {@code MongoClient} is created, it is configured with the same
+         * options as the parent {@code MongoClient} except {@code minPoolSize} is set to {@code 0}
+         * and {@code AutoEncryptionSettings} is omitted.
+         *
          * @param autoEncryptionSettings the auto-encryption settings
          * @return this
          * @since 3.11
          * @see #getAutoEncryptionSettings()
          */
-        public Builder autoEncryptionSettings(final AutoEncryptionSettings autoEncryptionSettings) {
+        public Builder autoEncryptionSettings(@Nullable final AutoEncryptionSettings autoEncryptionSettings) {
```

Comment:
Made this nullable so that auto-encryption could easily be disabled in the cloned settings.

Reply:
I'd be useful to also add the `@Nullable` annotation to the corresponding `MongoClientSettings.autoEncryptionSettings` field. Builder fields are expected to be nullable (builder objects are expected to exist in a partially constructed state) but for fields of other objects nullability is not so obvious.

It seems like `autoEncryptionSettings` was allowed to be null even before this PR, and a few other fields of the `MongoClientSettings` class are effectively nullable, but most of them are not.

Reply:
I'm not sure what you mean.  The getter for that field is already Nullable:

    @Nullable
    public AutoEncryptionSettings getAutoEncryptionSettings() {
        return autoEncryptionSettings;
    }

Reply:
I agree there is inconsistency in MongoClientSettings about this.  For example, if autoEncryptionSettings can accept null, why not streamFactoryFactory or credential?  I think we missed a use case when considering whether to allow null on these builder setters.  Since the default value of those fields are null, it seemed unnecessary to also accept null in the setter.  But what we're doing in the PR shows another use case: when you construct a builder from an existing settings, and need to null out some of the values from the settings.    So I think it would be best to allow null in the setters for all Nullable properties.  But let's do that in a separate PR than this.

Reply:
> The getter for that field is already Nullable

Yes, the getter is marked, I was thinking about additionally marking the field the getter returns, despite having the annotation on the getter is definitely helpful.

> I think it would be best to allow null in the setters for all Nullable properties. But let's do that in a separate PR than this.

Absolutely